### PR TITLE
[settings] Support functions

### PR DESCRIPTION
**Pull request by fvantienen**
Add function only settings

### CODE REVIEW

**Comment by fvantienen** [comment]
This has a merge conflict with the renaming of `mcu_periph_energy_save` to `mcu_energy_save` with #2859. Preferably this would be merged first, then I will rebase the other PR.

File: conf/settings/fixedwing_basic.xml [hunk 1]
```diff
@@ -23,5 +23,9 @@
       <dl_setting MAX="1" MIN="0" STEP="1" VAR="autopilot.kill_throttle"/>
     </dl_settings>
 
+    <dl_settings NAME="system">
```
Comment (gautierhattenberger) [inline conf/settings/fixedwing_basic.xml:26]: I suggest `mcu` name, there are already too many `system` tab in settings

File: sw/airborne/arch/chibios/mcu_arch.c [hunk 2]
```diff
@@ -162,5 +204,75 @@ void mcu_periph_energy_save(void)
     palClearLine(input_low);
   }
 #endif
```
Comment (gautierhattenberger) [inline sw/airborne/arch/chibios/mcu_arch.c:206]: a closing parenthesis is missing

File: sw/airborne/modules/boards/power_switch.c [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+ */
+void power_switch_set(bool val) {
+  if(val) {
+    gpio_set(POWER_SWITCH_GPIO);
+  } else {
+    gpio_clear(POWER_SWITCH_GPIO);
+  }
+
+  power_switch_status = val;
+}
```
Comment (gautierhattenberger) [inline sw/airborne/modules/boards/power_switch.c:61]: new line at end of file is missing, same for header

**Review by gautierhattenberger — CHANGES_REQUESTED** [review]

**Comment by fvantienen** [comment]
Should be ok to merge now

**Review by gautierhattenberger — APPROVED** [review]

**Pull request merged**